src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotValidateQueryStep.java

```diff
@@ -49,11 +49,6 @@ public StepResult doStep(FlightContext context) {
           StepStatus.STEP_RESULT_FAILURE_FATAL, new MismatchedValueException(message));
     }
 
-    if (datasetNames.size() > 1) {
-      String message = String.format("Snapshots can currently only be associated with one dataset");
-      return new StepResult(
-          StepStatus.STEP_RESULT_FAILURE_FATAL, new MismatchedValueException(message));
-    }
```

Comment:
Note for code reviewers: As far as I can tell, this is exactly duplicated with the if statement on lines 46-50.